### PR TITLE
Fix/compilation issues

### PR DESCRIPTION
**Pull request by plbossart**
we should really track such issues on a more systematic basis.

I found this out by looking at Travis results with W=1

@xiulipan is there a way to catch this earlier in CI?

### CODE REVIEW

**Comment by plbossart** [comment]
updated with additional fixes, and a very nasty one letter bug identified.
We really need to reach a zero warning goal so that serious issues can be reported instead of being buried in heaps of straightforward/minor problems

File: sound/usb/mixer_quirks.c [hunk 1]
```diff
@@ -1691,7 +1691,7 @@ static const struct snd_kcontrol_new snd_microii_mixer_spdif[] = {
 static int snd_microii_controls_create(struct usb_mixer_interface *mixer)
 {
 	int err, i;
-	const static usb_mixer_elem_resume_func_t resume_funcs[] = {
+	static const usb_mixer_elem_resume_func_t resume_funcs[] = {
```
Comment (lyakh) [inline sound/usb/mixer_quirks.c:1694]: actually `static` isn't needed here and it probably doesn't even have many advantages - this function is called from `snd_usb_mixer_apply_create_quirk()` only for one specific USB ID. So only in a very unlikely case of someone having multiple such USB cards or replugging one such card multiple times will this function be called multiple times. Only in such a case having an array `static` might have an advantage. Better just remove `static`

File: sound/soc/intel/boards/kbl_rt5660.c [hunk 2]
```diff
@@ -138,13 +138,13 @@ static int kabylake_ssp0_fixup(struct snd_soc_pcm_runtime *rtd,
 {
 	struct snd_interval *rate = hw_param_interval(params,
 			SNDRV_PCM_HW_PARAM_RATE);
-	struct snd_interval *channels = hw_param_interval(params,
+	struct snd_interval *chan = hw_param_interval(params,
```
Comment (lyakh) [inline sound/soc/intel/boards/kbl_rt5660.c:141]: in fact IMHO even file-static variables with such generic names like `channels` or `rates` aren't a very good thing. One reason is the problem, that we get here, another reason is, that if many files (in this case board drivers) have variables with the same name, searching for them using grep or (vi or emacs) tags becomes difficult. Ideally I'd rename those variables to be file-specific (e.g. `kabylake_rt5660_channels` just like `kabylake_rt5660_ops` that's already used there), but that's more work of course...

File: sound/usb/mixer_quirks.c (hunk 1 above)
Comment (plbossart) [reply sound/usb/mixer_quirks.c:1694]: merged upstream already

**Comment by xiulipan** [comment]
@plbossart I can create some new item in the CI check list looks like:
`sof-ci/jenkins/pr-checkwaring`
is that OK for you with this?

**Comment by plbossart** [comment]
> @plbossart I can create some new item in the CI check list looks like:
> `sof-ci/jenkins/pr-checkwaring`
> is that OK for you with this?

sure. With all the cleanups there should not be any warnings in sound or soundwire.

**Pull request merged**